unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -32,6 +33,23 @@
   public static final int LONG_ARRAY_OFFSET;
 
   public static final int DOUBLE_ARRAY_OFFSET;
+  
+  private static final boolean unaligned;
+  static {
+    // Copied from java.nio.Bits.java#unaligned
```

Comment:
Is there any license issue here? I'm not sure. But maybe we can use reflection to call `Bits.unaligned` and add a test for it in case `Bits.unaligned` is removed in future.